src/include/homestore/btree/btree.hpp

```diff
@@ -64,13 +64,22 @@ class Btree {
 
     // This workaround of BtreeThreadVariables is needed instead of directly declaring statics
     // to overcome the gcc bug, pointer here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66944
+//    static BtreeThreadVariables* bt_thread_vars() {
+//        static thread_local BtreeThreadVariables* s_ptr{nullptr};
+//        if (s_ptr == nullptr) {
+//            static thread_local BtreeThreadVariables inst;
+//            s_ptr = &inst;
+//        }
+//        return s_ptr;
+//    }
     static BtreeThreadVariables* bt_thread_vars() {
-        static thread_local BtreeThreadVariables* s_ptr{nullptr};
-        if (s_ptr == nullptr) {
-            static thread_local BtreeThreadVariables inst;
-            s_ptr = &inst;
-        }
-        return s_ptr;
+        //static boost::fibers::local_fiber <BtreeThreadVariables> *s_ptr{nullptr};
+        auto this_id( boost::this_fiber::get_id() );
+        static thread_local std::map<fibers::fiber::id,  std::unique_ptr<BtreeThreadVariables>> fiber_map;
```

Comment:
Does this assume that fibers are not moved across threads. In that case fiber1 will have entry in multiple threads.